src/main/java/com/laytonsmith/abstraction/MCTagContainer.java

```diff
@@ -0,0 +1,67 @@
+package com.laytonsmith.abstraction;
+
+import com.laytonsmith.abstraction.enums.MCTagType;
+
+import java.util.Collection;
+
+/**
+ * Minecraft NBT containers that can be used to read and modify tags in supported game objects.
+ * This includes item meta, entities, block entities, chunks, worlds, etc.
+ */
+public interface MCTagContainer extends AbstractionObject {
+
+	/**
+	 * Returns whether the tag container does not contain any tags.
+	 * @return whether container is empty
+	 */
+	boolean isEmpty();
+
+	/**
+	 * Gets a set of key objects for each tag that exists in this container.
+	 * These are minecraft formatted namespaced keys. (e.g. "namespace:key")
+	 * These key objects can be passed to other methods in this class.
+	 * @return a set of keys
+	 */
+	Collection getKeys();
```

Comment:
Collection of what? I guess object, but would be good to be explicit about it, `Collection<Object>`. Also, I'm starting to think that it would be better to have a wrapper class here, even if internally it just has the one Object field. That way, it's at least encapsulated to make it less error prone at the top level.

Reply:
I was torn on this, but thought I'd see if it made sense to you, since we only ever use the string representation. (it's a string key that bukkit converts into an object that splits the namespace and key and then we convert back to a string, which felt wasteful) Anyway, updated.

Reply:
Yeah, preventing boxing feels like premature optimization. The additional clarity from ensuring correctness seems like it's worth more in this case.